include/linalg.hpp

```diff
@@ -723,12 +723,34 @@ namespace cytnx {
                                                const cytnx_uint64 &keepdim, const double &err = 0.,
                                                const bool &is_UvT = true,
                                                const unsigned int &return_err = 0,
-                                               const unsigned int &mindim = 1);
+                                               const cytnx_uint64 &mindim = 1);
+
+    /**
+     * @brief Perform Singular-Value decomposition on a UniTensor with truncation and keep at most
+     * \p minblockdim singular values in each block.
+     * @details This function performs the Singular-Value decomposition of a UniTensor \p Tin and
+     * truncates the singular values. The result will depend on the rowrank of the
+     * UniTensor \p Tin. For each block, the minimum dimension can be chosen. This can be helpful to
+     * avoid loosing symmetry sectors in the truncated SVD. For more details, please refer to the
+     * documentation of the function Svd_truncate(const Tensor &Tin, const cytnx_uint64 &keepdim,
+     * const double &err, const bool &is_UvT, const unsigned int &return_err).
+     * @param[in] minblockdim a vector containing the minimum dimension of each block;
+     * alternatively, a vector with only one element can be given to have the same minblockdim for
+     * each block
+     * @see Svd_truncate(const Tensor &Tin, const cytnx_uint64 &keepdim, const double &err,
+     *                   const bool &is_UvT, const unsigned int &return_err)
+     */
+    std::vector<cytnx::UniTensor> Svd_truncate(const cytnx::UniTensor &Tin,
+                                               const cytnx_uint64 &keepdim,
+                                               const std::vector<cytnx_uint64> minblockdim,
```

Comment:
If we are making a copy, we may remove `const`. If not, we may pass it as a const reference.